Poker.BE/Poker.BE.Domain/Core/User.cs

```diff
@@ -9,138 +10,154 @@
+        public Player EnterRoom(Room room)
+        {
+            // check that the user doesn't have already a player in the room
+            if (Players.Count > 0 && room.Players.Count > 0)
+            {
+                var result = from player in room.Players
+                             where Players.Contains(player, new Utility.AddressComparer<Player>())
+                             select player;
+
+                if (result.Count() > 0)
+                {
+                    throw new RoomRulesException(
+                        string.Format("the user already play this room: {0} with player nickname: {1}",
+                        room.Name, result.First().Nickname));
+                }
+            }
+
+            // entering the room
+            var freshPlayer = gameCenter.EnterRoom(room);
+
+            // logging
+            logger.Log(string.Format("User {0} has player {1}", UserName, freshPlayer.GetHashCode()),
+                this, "Medium");
+
+            //registering to an event: the player got the turn
+            freshPlayer.TurnChanged += ChoosePlayMove;
+
+            //Adding the player to players collection
+            Players.Add(freshPlayer);
+
+            return freshPlayer;
+        }
+
+        public void ChoosePlayMove(object sender, EventArgs e)
+        {
+            //Note: systactic suger for checking if sender is a player type
+            if (sender is Player player && e is PlayMoveEventArgs playMoveEvent)
+            {
+                if (!Players.Contains(player))
+                    throw new PlayerNotFoundException("This player doesn't belong to this user");
+
+                player.ChoosePlayMove(playMoveEvent.PlayMove, playMoveEvent.AmountToBetOrCall);
+            }
+        }
+
+        public Room CreateNewRoom(int level, GameConfig config, out Player creator)
+        {
+            var result = gameCenter.CreateNewRoom(level, config, out creator);
+            Players.Add(creator);
+
+            // log info
+            logger.Info(string.Format("user {0} has created an new room {1}", GetHashCode(), result.GetHashCode()), this);
+
+            return result;
+        }
+
+        public void JoinNextHand(Player player, int seatIndex, double buyIn)
+        {
+            if (!Players.Contains(player, new Utility.AddressComparer<Player>()))
+            {
+                throw new PlayerNotFoundException("the user doesn't have this player");
+            }
+
+            gameCenter.JoinNextHand(player, seatIndex, buyIn);
+        }
+
+        public double StandUpToSpactate(Player player)
+        {
+            if (!Players.Contains(player, new Utility.AddressComparer<Player>()))
+            {
+                throw new PlayerNotFoundException("the user doesn't have this player");
+            }
+
+            UserBank.Money = gameCenter.StandUpToSpactate(player);
+
+            return UserBank.Money;
+        }
+        #endregion
+
+
+        #endregion
+
+    }
```

Comment:
❓ the `git diff` here is massive so it's hard to see what had changed. do you remember?